<ui version="4.0" >
 <author>Marco Clemencic</author>
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>965</width>
    <height>655</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string comment="Browser for LHCb Conditions Database" >CondDBBrowser</string>
  </property>
  <property name="windowIcon" >
   <iconset resource="../../../qt_resources/Resources.qrc" >
    <normaloff>:/icons/icon.png</normaloff>:/icons/icon.png</iconset>
  </property>
  <widget class="QWidget" name="mainPanel" >
   <layout class="QVBoxLayout" name="mainPanelLayout" >
    <property name="margin" >
     <number>9</number>
    </property>
    <item>
     <widget class="QSplitter" name="mainWidget" >
      <property name="orientation" >
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" native="1" name="browsePanel" >
       <layout class="QVBoxLayout" name="browsePanelLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <item>
         <widget class="QWidget" native="1" name="pathPanel" >
          <layout class="QHBoxLayout" name="horizontalLayout" >
           <property name="margin" >
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="pathLabel" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Preferred" hsizetype="Minimum" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text" >
              <string>Path</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="pathComboBox" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="editable" >
              <bool>false</bool>
             </property>
             <property name="insertPolicy" >
              <enum>QComboBox::NoInsert</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QTreeView" name="hierarchyTreeView" >
          <property name="editTriggers" >
           <set>QAbstractItemView::NoEditTriggers</set>
          </property>
          <property name="showDropIndicator" stdset="0" >
           <bool>false</bool>
          </property>
          <property name="headerHidden" >
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QTextEdit" name="dataView" >
       <property name="acceptDrops" >
        <bool>false</bool>
       </property>
       <property name="undoRedoEnabled" >
        <bool>false</bool>
       </property>
       <property name="acceptRichText" >
        <bool>false</bool>
       </property>
       <property name="textInteractionFlags" >
        <set>Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
       </property>
      </widget>
      <widget class="QWidget" native="1" name="filterPanel" >
       <layout class="QVBoxLayout" name="filterPanelLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <item>
         <widget class="QGroupBox" name="iovTagFilterGroupBox" >
          <property name="sizePolicy" >
           <sizepolicy vsizetype="Minimum" hsizetype="Preferred" >
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title" >
           <string>Filter</string>
          </property>
          <layout class="QFormLayout" name="iovTagFilterLayout" >
           <property name="fieldGrowthPolicy" >
            <enum>QFormLayout::ExpandingFieldsGrow</enum>
           </property>
           <property name="labelAlignment" >
            <set>Qt::AlignBottom|Qt::AlignRight|Qt::AlignTrailing</set>
           </property>
           <property name="formAlignment" >
            <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
           </property>
           <item row="0" column="0" >
            <widget class="QLabel" name="sinceLabel" >
             <property name="text" >
              <string>From time</string>
             </property>
            </widget>
           </item>
           <item row="1" column="0" >
            <widget class="QLabel" name="untilLabel" >
             <property name="text" >
              <string>To time</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0" >
            <widget class="QLabel" name="tagLabel" >
             <property name="text" >
              <string>Tag name</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1" >
            <widget class="QComboBox" name="tagComboBox" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item row="3" column="1" >
            <widget class="QCheckBox" name="hideAutoCheckBox" >
             <property name="toolTip" >
              <string>If checked, the special tags starting with "_auto_" are not displayed.</string>
             </property>
             <property name="text" >
              <string>Hide "_auto_" tags</string>
             </property>
             <property name="checked" >
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="0" column="1" >
            <widget class="TimePointEdit" native="1" name="sinceFilterWidget" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Minimum" hsizetype="MinimumExpanding" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item row="1" column="1" >
            <widget class="TimePointEdit" native="1" name="untilFilterWidget" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Minimum" hsizetype="MinimumExpanding" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="iovTableGroupBox" >
          <property name="sizePolicy" >
           <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title" >
           <string>IOV Table</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout" >
           <item>
            <widget class="QTableView" name="iovView" >
             <property name="enabled" >
              <bool>true</bool>
             </property>
             <property name="editTriggers" >
              <set>QAbstractItemView::NoEditTriggers</set>
             </property>
             <property name="selectionMode" >
              <enum>QAbstractItemView::SingleSelection</enum>
             </property>
             <property name="selectionBehavior" >
              <enum>QAbstractItemView::SelectRows</enum>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="iovUTCCheckBox" >
             <property name="toolTip" >
              <string>Show the IoVs as UTC</string>
             </property>
             <property name="text" >
              <string>UTC</string>
             </property>
             <property name="checked" >
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="payloadGroupBox" >
          <property name="sizePolicy" >
           <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title" >
           <string>Payload</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2" >
           <item>
            <widget class="QListView" name="fieldsView" />
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>0</y>
     <width>965</width>
     <height>29</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuDataBase" >
    <property name="title" >
     <string>DataBase</string>
    </property>
    <widget class="QMenu" name="menuRecent" >
     <property name="title" >
      <string>Recent</string>
     </property>
    </widget>
    <widget class="QMenu" name="menuStandard" >
     <property name="title" >
      <string>Standard</string>
     </property>
    </widget>
    <addaction name="actionNew" />
    <addaction name="actionOpen" />
    <addaction name="menuRecent" />
    <addaction name="menuStandard" />
    <addaction name="separator" />
    <addaction name="actionRead_Only" />
    <addaction name="separator" />
    <addaction name="actionSlice" />
    <addaction name="actionDump_to_files" />
    <addaction name="actionClose" />
    <addaction name="separator" />
    <addaction name="actionQuit" />
   </widget>
   <widget class="QMenu" name="menuEdit" >
    <property name="title" >
     <string>Edit</string>
    </property>
    <addaction name="actionAdd_Condition" />
    <addaction name="actionNew_Tag" />
   </widget>
   <widget class="QMenu" name="menuAdvanced" >
    <property name="enabled" >
     <bool>true</bool>
    </property>
    <property name="title" >
     <string>Advanced</string>
    </property>
    <addaction name="actionNew_Node" />
    <addaction name="actionDelete_Node" />
    <addaction name="actionDelete_Tag" />
   </widget>
   <widget class="QMenu" name="menuHelp" >
    <property name="title" >
     <string>Help</string>
    </property>
    <addaction name="actionAbout" />
    <addaction name="actionAbout_Qt" />
   </widget>
   <addaction name="menuDataBase" />
   <addaction name="menuEdit" />
   <addaction name="menuAdvanced" />
   <addaction name="menuHelp" />
  </widget>
  <widget class="QStatusBar" name="statusbar" />
  <action name="actionNew" >
   <property name="text" >
    <string>New</string>
   </property>
  </action>
  <action name="actionOpen" >
   <property name="text" >
    <string>Open</string>
   </property>
  </action>
  <action name="actionSlice" >
   <property name="text" >
    <string>Slice</string>
   </property>
  </action>
  <action name="actionClose" >
   <property name="text" >
    <string>Close</string>
   </property>
  </action>
  <action name="actionAdd_Condition" >
   <property name="text" >
    <string>Add Condition</string>
   </property>
  </action>
  <action name="actionNew_Tag" >
   <property name="text" >
    <string>New Tag</string>
   </property>
  </action>
  <action name="actionNew_Node" >
   <property name="text" >
    <string>New Node</string>
   </property>
  </action>
  <action name="actionDelete_Node" >
   <property name="text" >
    <string>Delete Node</string>
   </property>
  </action>
  <action name="actionDelete_Tag" >
   <property name="text" >
    <string>Delete Tag</string>
   </property>
  </action>
  <action name="actionAbout" >
   <property name="text" >
    <string>About</string>
   </property>
  </action>
  <action name="actionQuit" >
   <property name="text" >
    <string>Quit</string>
   </property>
  </action>
  <action name="actionAbout_Qt" >
   <property name="text" >
    <string>About Qt</string>
   </property>
  </action>
  <action name="actionDump_to_files" >
   <property name="text" >
    <string>Dump to files</string>
   </property>
  </action>
  <action name="actionRead_Only" >
   <property name="checkable" >
    <bool>true</bool>
   </property>
   <property name="checked" >
    <bool>true</bool>
   </property>
   <property name="enabled" >
    <bool>false</bool>
   </property>
   <property name="text" >
    <string>Read Only</string>
   </property>
   <property name="statusTip" >
    <string>Change the opening mode of the database (re-opens it)</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TimePointEdit</class>
   <extends>QWidget</extends>
   <header>CondDBUI/Browser/Widgets.h</header>
   <container>1</container>
   <slots>
    <slot>setMinimumDateTime(QDateTime)</slot>
    <slot>setMaximumDateTime(QDateTime)</slot>
    <slot>resetMinimumDateTime()</slot>
    <slot>resetMaximumDateTime()</slot>
    <signal>dateTimeChanged(QDateTime)</signal>
    <signal>dateChanged(QDate)</signal>
    <signal>timeChanged(QTime)</signal>
    <signal>maxChanged(bool)</signal>
    <signal>maxChecked()</signal>
    <signal>maxUnchecked()</signal>
    <signal>validityKeyChanged(cool::ValidityKey)</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../../qt_resources/Resources.qrc" />
 </resources>
 <connections>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>451</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout_Qt</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>aboutQt()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>451</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionClose</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>closeDatabase()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>451</x>
     <y>336</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>aboutDialog()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>482</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pathComboBox</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>MainWindow</receiver>
   <slot>selectedPath(QString)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>187</x>
     <y>48</y>
    </hint>
    <hint type="destinationlabel" >
     <x>482</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionRead_Only</sender>
   <signal>triggered(bool)</signal>
   <receiver>MainWindow</receiver>
   <slot>reopenDatabase(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel" >
     <x>482</x>
     <y>327</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sinceFilterWidget</sender>
   <signal>dateTimeChanged(QDateTime)</signal>
   <receiver>untilFilterWidget</receiver>
   <slot>setMinimumDateTime(QDateTime)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>851</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel" >
     <x>851</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>untilFilterWidget</sender>
   <signal>maxChecked()</signal>
   <receiver>sinceFilterWidget</receiver>
   <slot>resetMaximumDateTime()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>851</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel" >
     <x>851</x>
     <y>72</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>untilFilterWidget</sender>
   <signal>dateTimeChanged(QDateTime)</signal>
   <receiver>sinceFilterWidget</receiver>
   <slot>setMaximumDateTime(QDateTime)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>851</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel" >
     <x>851</x>
     <y>72</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>aboutQt()</slot>
  <slot>openStandardDatabase()</slot>
  <slot>closeDatabase()</slot>
  <slot>aboutDialog()</slot>
  <slot>selectedItem(QModelIndex)</slot>
  <slot>selectedPath(QString)</slot>
  <slot>testSlot()</slot>
  <slot>reopenDatabase(bool)</slot>
  <slot>sinceChanged(QDateTime)</slot>
  <slot>untilChanged(QDateTime)</slot>
  <signal>openedDB(P*)</signal>
 </slots>
</ui>
